src/components/Header/HeaderNavItem/HeaderNavItem.tsx

```diff
@@ -0,0 +1,59 @@
+import styles from "./HeaderNavItem.module.css";
+import classNames from "classnames";
+import type {
+  NavBarItem,
+  SubMenuObject,
+} from "../../../constants/Navbar.constants";
+import type { JSX } from "react";
+
+type HeaderNavItemProps = NavBarItem;
+
+const HeaderNavItem = (props: HeaderNavItemProps) => {
+  return (
+    <div className={styles.navbar_link}>
+      <span
+        className={classNames(styles.navbar_item, {
+          [styles.arrow]: props.isArrow,
+        })}
+      >
+        {props.title}
+      </span>
+      {props.subMenuObjects && (
+        <ul className={styles.navbar_dropdown_content}>
+          {props.subMenuObjects.map((group, groupIndex) => (
+            <li key={groupIndex} className={styles.navbar_group}>
+              {group.map((item) => (
+                <LinkObject
+                  key={item.text}
+                  text={item.text}
+                  href={item.href}
+                  extraContent={item.extraContent}
+                />
+              ))}
+            </li>
+          ))}
+        </ul>
+      )}
+    </div>
+  );
+};
+type LinkObjectProps = SubMenuObject;
+
+const LinkObject = (el : LinkObjectProps) => {
+  function isJSXElement(value: unknown): value is () => JSX.Element {
+    return typeof value === 'function';
+  }
+  const content = (
+    <>
+      {el.text} {isJSXElement(el.extraContent) && <el.extraContent />}
```

Comment:
rewrite type from unknown to ()=> JSX.Element in constants and remove this type checker. also add useMemo